Modules/_datetimemodule.c

```diff
@@ -2986,6 +2986,61 @@ date_fromisoformat(PyObject *cls, PyObject *dtstr)
     return NULL;
 }
 
+
+static int _iso_long_year_helper(int year) {
+    return ((year + (year / 4) - (year / 100) + (year / 400)) % 7);
+}
+
+static PyObject *
+date_fromisocalendar(PyObject *cls, PyObject *args, PyObject *kw) {
+    static char *keywords[] = {
+        "year", "week", "day", NULL
+    };
+
+    int year, week, day;
+    if (PyArg_ParseTupleAndKeywords(args, kw, "iii:fromisocalendar",
+                keywords,
+                &year, &week, &day) == 0) {
+        if (PyErr_ExceptionMatches(PyExc_OverflowError)) {
+            PyErr_Format(PyExc_ValueError,
+                    "ISO calendar component out of range");
+
+        }
+        return NULL;
+    }
+
+    // Year is bounded to 0 < year < 10000 because 9999-12-31 is (9999, 52, 5)
+    if (year <= 0 || year >= 10000) {
+        PyErr_Format(PyExc_ValueError, "Year is out of range: %d", year);
+        return NULL;
+    }
+
+    if (week <= 0 || week >= 53) {
+        if (!(week == 53 &&
+             (_iso_long_year_helper(year) == 4 ||
+              _iso_long_year_helper(year - 1) == 3))) {
+            PyErr_Format(PyExc_ValueError, "Invalid week: %d", week);
+            return NULL;
+        }
+    }
+
+    if (day <= 0 || day >= 8) {
+        PyErr_Format(PyExc_ValueError, "Invalid day: %d (range is [1, 7])");
```

Comment:
It looks like you're missing an argument.

Reply:
Oof. Now I'm tempted to add a test for this, even though I am generally against the idea of testing that error messages say a specific thing.

I think I'll just fix it and leave it untested. Good catch!